### ISSUE TITLE
Migrate bootstrap.js into the WebExtension

### ISSUE THREAD

**Issue by Standard8** [protz/thunderbird-conversations#1423]
bootstrap.js is still running via the legacy add-on paths. We should migrate this so that the equivalent calls are run from background.js.

The first part of this is #1422. 

The window observers I'd like to move to being WebExtension observers, and then pass new windows to the experiment API which does the necessary monkey patching.

**Comment by Standard8** [comment]
I've now got a way of doing this that allows us to still use the existing chrome registration, but avoid needing to be marked as a legacy add-on and also let us migrate bootstrap into the experimental API.

This is basically ready to go, but I'm hoping to do two things at the same time and get compatibility with the current beta going. That means more string translations, but I think these will all be in addon/content/modules and hopefully won't affect @siefkenj.

**Comment by Standard8** [comment]
@siefkenj I've just landed this work. Note I've just updated the development doc, especially about specifying `-purgecaches` to Thunderbird:

https://github.com/protz/thunderbird-conversations/blob/master/docs/Development.md

I'm hoping that's something we can eradicate over time as we move away from modules and more to experiment or real APIs. I might also be able to improve it with a few more code fixes, but I'll have to test it out a bit more.
